### DISCUSSION TITLE
Why is `just` recommended over mission-control?

### DISCUSSION THREAD

**Discussion by s3igo** [Platonic-Systems/mission-control#41] [Q&A] [answered]
In the README, it states:

> We recommend that you use just over this module.

I believe this is a great project, particularly because it has two advantages over just: the ability to work entirely within Nix, and the integration with pkgs.writeShellApplication, allowing the use of ShellCheck.

However, I’m curious to know why just is being recommended instead of this project. What are the reasons behind this recommendation?

**Comment by srid** [comment, marked as answer]
You can still use `pkgs.writeShellApplication` when using `just`, by adding it to either `packages` or `apps` flake output, and then running `nix run` from the `justfile` accordingly. Alternatively, you can put those scripts in your devShells's `nativeBuildInputs`, and then directly use them in `justfile`, if you are using `just` always from the devShell.

`just` also provides shell completion, but mission-control does not (#4).

**Reply by s3igo** [reply]
While using `pkgs.writeShellApplication` and the `apps` output in flake is convenient, I find myself having to write a lot of boilerplate, which adds unnecessary visual noise. That's why I prefer the approach taken by mission-control, as it helps reduce this clutter.

If the only difference between mission-control and just is shell completion, does that mean it's simply a matter of personal preference?
Personally, I think that once #31 is resolved, #4 can be resolved as well.

**Reply by srid** [reply]
> If the only difference between mission-control and just is shell completion, does that mean it's simply a matter of personal preference?

There's a matter of personal preference, yes. But there are objective differences as well, due to the features inherent to `just` itself (modules, etc.). 

Anyway, I'm open to keeping this project alive, if anybody would like to contribute PRs (towards #31 and #4 for example), and maybe even become a maintainer down the line. It is just that I, as a maintainer, have moved to `just` for all my projects.

**Reply by s3igo** [reply]
I see, I'm glad to know that mission-control isn't being deprecated. Thank you for answering my question!
